duckduckgo/iOS
# Add https lookup service

Task/Issue URL: https://app.asana.com/0/inbox/361428290920650/1132869367787864/1136983228981334
Tech Design URL: https://app.asana.com/0/inbox/361428290920650/890441675382858/1136983228981334

**Description**:
Add a https lookup service

**Steps to test this PR**:

**Test item in local list is upgraded**
1. Visit http://facebook.com
1. Check logs "facebook.com is locally upgradable"
1. Check that https local lookup pixel "m_https_lu" is sent
1. Note this request subsequently loads https and redirects so will also trigger some "m_https_nl" pixels will also fire

**Test when item not in local list but in service then is upgraded and correct pixels are sent****
1. Visit http://channel4.com
1. Check logs "channel4.com is not locally upgradable" and "channel4.com is service upgradable"
1. Check that https service request pixel "m_https_sru" is sent
1. Note this request subsequently loads https and redirects so will also trigger some "m_https_nl" pixels will also fire

**Test when item not in local list but in service cache, it is upgraded and correct pixels are sent**
1. Visit http://channel4.com again
1. Check logs "channel4.com is not locally upgradable" and "channel4.com is service upgradable"
1. Check that this time the https service request pixel "m_https_scu" is sent
1. Note this request subsequently loads https and redirects so will also trigger some "m_https_nl" pixels will also fire

**Test when item not in local list or in service list, it is not upgraded and correct pixels are sent**
1. Visit http://example.com
1. Check logs "example.com is not locally upgradable" and "example.com is not service upgradable"
1. Check that the https service not upgraded request pixel "m_https_srn" is sent

**Test when item not in local list and not in service cache, it is not upgraded and correct pixels are sent**
1. Visit http://example.com again
1. Check logs "example.com is not locally upgradable" and "example.com is not service upgradable"
1. Check that this time the https service not upgraded cache pixel "m_https_scn" is sent

**Test when item in whitelist then is not upgraded and correct pixels are sent****
1. Visit http://www.bjtax.gov.cn
1. Check that page is not upgraded
1. Check that https no lookup pixel "m_https_nl" pixel is sent

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
